crates/walrus-sdk/src/client.rs

```diff
@@ -180,15 +180,35 @@ impl WalrusNodeClient<()> {
         config: ClientConfig,
         committees_handle: CommitteesRefresherHandle,
     ) -> ClientResult<Self> {
-        Self::new_inner(config, committees_handle, None, None).await
+        let (encoding_config, communication_limits) =
+            Self::build_encoding_artifacts(&config, committees_handle.n_shards());
+        Self::new_inner(
+            config,
+            committees_handle,
+            encoding_config,
+            communication_limits,
+            None,
+            None,
+        )
+        .await
     }
     /// Creates a new Walrus client without a Sui client.
     pub async fn new_with_max_blob_size(
         config: ClientConfig,
         committees_handle: CommitteesRefresherHandle,
         max_blob_size: Option<u64>,
     ) -> ClientResult<Self> {
-        Self::new_inner(config, committees_handle, None, max_blob_size).await
+        let (encoding_config, communication_limits) =
+            Self::build_encoding_artifacts(&config, committees_handle.n_shards());
+        Self::new_inner(
+            config,
+            committees_handle,
+            encoding_config,
+            communication_limits,
+            None,
+            max_blob_size,
+        )
+        .await
```

Comment:
Question: Why do we need the separate `build_encoding_artifacts` function? Could we not do this in the `new_inner` function?